tasks/main.yaml

```diff
@@ -3,16 +3,39 @@
 - name: Test if distribution is RedHat
   assert:
     that: >
-      ansible_os_family == "RedHat" and
-      ansible_distribution_major_version | int >= 6
+      (ansible_os_family == "RedHat" and
+      ansible_distribution_major_version | int >= 6)
+      or (ansible_distribution == "Amazon" and
+      ansible_distribution_version == "2015.09")
 
-- name: Install rabbitmq-server
+- name: Install erlang from yum repos
+  yum:
+    name: "{{ erlang_pkg }}"
+  notify:
+    - Restart rabbitmq-server
+  tags:
+    - erlang_pkg
+
+- name: Install rabbitmq-server from yum repos
   yum:
     name: "{{ rabbitmq_pkg }}"
   notify:
     - Restart rabbitmq-server
   tags:
     - rabbitmq_pkg
+  when: rabbitmq_yum_repo_url is undefined
+
+- name: Add Rabbit signing-key
+  rpm_key: state=present key=https://www.rabbitmq.com/rabbitmq-signing-key-public.asc
```

Comment:
Do not use the one-line Ansible parameters (`state=present`). Use pure YAML instead (`state: present`).